### PR TITLE
Add Flawfinder Parser

### PR DESCRIPTION
**Pull request by art-of-dom**
This change adds a Flawfinder parser which is a static analysis tool for finding potential security flaws in C/C++.  Project can be found here: https://github.com/david-a-wheeler/flawfinder

### CODE REVIEW

File: src/main/java/edu/hm/hafner/analysis/parser/FlawfinderParser.java [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+
+/**
+ * A parser for the flawfinder warnings.
+ *
+ * @author Dom Postorivo
+ */
+public class FlawfinderParser extends RegexpLineParser {
+    private static final long serialVersionUID = 8088991846076174837L;
+
+    private static final String FLAWFINDER_WARNING_PATTERN = "^(.*):(\\d+): .\\[([012345])\\] \\(([a-z0-9]*)\\) (.*)$";
```
Comment (uhafner) [inline src/main/java/edu/hm/hafner/analysis/parser/FlawfinderParser.java:19]: Can you please use named groups? This makes the regex and matchers better readable, see [EclipseParser.java#L28](https://github.com/jenkinsci/analysis-model/blob/master/src/main/java/edu/hm/hafner/analysis/parser/EclipseParser.java#L28) for an example

Comment (art-of-dom) [reply src/main/java/edu/hm/hafner/analysis/parser/FlawfinderParser.java]: Done. I was using the RuboCopParser as an example. If I have time I may go back and clean it up.

**Comment by uhafner** [comment]
Thanks!

**Comment by uhafner** [comment]
The new parser is now part of the 8.2.0 release: https://plugins.jenkins.io/analysis-model-api/.

If you like, you can now register the parser in the warnings plugin, here is an [example](https://github.com/jenkinsci/warnings-ng-plugin/blob/master/plugin/src/main/java/io/jenkins/plugins/analysis/warnings/EsLint.java). There you can define an icon, a help text, default pattern, etc.

**Pull request merged**